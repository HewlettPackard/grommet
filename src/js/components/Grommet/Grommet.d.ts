import * as React from 'react';
import Grommet from '../index';

export declare namespace GrommetTypes {
  type Direction = 'rtl' | 'ltr';
}

export interface GrommetProps extends Grommet.Props {
  dir: GrommetTypes.Direction;
  theme: {};
}

export class Grommet extends React.Component<GrommetProps, undefined> { }

export default Grommet;
