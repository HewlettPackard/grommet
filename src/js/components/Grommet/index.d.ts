import * as React from "react";

export interface GrommetProps {
  full?: boolean;
  theme?: object;
}

declare const Grommet: React.StatelessComponent<GrommetProps>;

export { Grommet };
