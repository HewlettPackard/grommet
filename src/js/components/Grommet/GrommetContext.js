import React from 'react';

// Track which components mounted onto the DOM are coming from
// Grommet (see FocusedContainer.js)
export const GrommetContext = React.createContext(undefined);
