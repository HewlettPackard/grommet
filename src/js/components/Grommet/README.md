## Grommet
This is the top level Grommet container.

[![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-site?initialpath=grommet&amp;module=%2Fscreens%2FGrommet.js)
## Usage

```javascript
import { Grommet } from 'grommet';
<Grommet>...</Grommet>
```

## Properties

**dir**

Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.

```
rtl
ltr
```

**theme**

Custom styles for Grommet app component.

```
object
```
  