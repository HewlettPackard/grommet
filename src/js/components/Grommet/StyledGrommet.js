import styled, { css } from 'styled-components';

import { baseStyle } from '../../utils';

const fullStyle = css`
  width: 100vw;
  height: 100vh;
`;

const StyledGrommet = styled.div`
  ${baseStyle}
  ${props => props.full && fullStyle}
  ${props => props.theme.global.font.face}
`;

export default StyledGrommet.extend`
  ${props => props.theme.grommet.extend}
`;
