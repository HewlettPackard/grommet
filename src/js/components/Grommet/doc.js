import { describe, PropTypes } from 'react-desc';

import { getAvailableAtBadge } from '../../utils';

export default (Grommet) => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet'))
    .description('This is the top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`
    );

  DocumentedGrommet.propTypes = {
    dir: PropTypes.oneOf(['rtl', 'ltr']).description(
      `Whether text should be rendered right to left or not. Defaults to
      inherit from the document context.`
    ),
    theme: PropTypes.object.description('Custom styles for Grommet app component.'),
  };

  return DocumentedGrommet;
};
