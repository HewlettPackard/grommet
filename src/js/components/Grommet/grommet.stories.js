import React from 'react';
import { storiesOf } from '@storybook/react';

import { Grommet, Anchor, Box } from 'grommet';
import { Add } from 'grommet-icons';

const customTheme = {
  global: {
    colors: {
      custom: '#cc6633',
    },
  },
};


const Themed = () => (
  <Grommet theme={customTheme}>
    <Box pad='medium'>
      <Anchor icon={<Add />} label='Add' color='custom' />
    </Box>
  </Grommet>
);

storiesOf('Grommet', module)
  .add('Theme', () => <Themed />);
