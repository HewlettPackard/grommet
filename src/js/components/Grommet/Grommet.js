import React, { forwardRef, useEffect, useMemo, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const deviceResponsive = (userAgent, theme) => {
  // log('--deviceResponsive', userAgent, theme);
  /*
   * Regexes provided for mobile and tablet detection are meant to replace
   * a full-featured specific library due to contributing a considerable size
   * into the bundle.
   *
   * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
   */
  if (userAgent) {
    if (/(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)) {
      return getDeviceBreakpoint('tablet', theme);
    }
    if (/Mobile|iPhone|Android/.test(userAgent)) {
      return getDeviceBreakpoint('phone', theme);
    }
    return getDeviceBreakpoint('computer', theme);
  }
  return undefined;
};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    ...rest
  } = props;

  const { background, dir, themeMode, userAgent } = props;

  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';
    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(window.innerWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <ContainerTargetContext.Provider value={containerTarget}>
          <StyledGrommet full={full} {...rest} ref={ref}>
            {children}
          </StyledGrommet>
          {full && <FullGlobalStyle />}
        </ContainerTargetContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
