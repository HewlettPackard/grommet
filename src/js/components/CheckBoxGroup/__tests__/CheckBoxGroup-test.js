import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Grommet } from '../../Grommet';
import { CheckBoxGroup } from '..';

describe('CheckBoxGroup', () => {
  afterEach(cleanup);

  test('options renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup options={['First', 'Second']} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('checked renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup checked={['First']} options={['First', 'Second']} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('checked on options renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup
          options={[
            { label: 'Maui', checked: true },
            { label: 'Jerusalem' },
            { label: 'Wuhan' },
          ]}
        />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('disabled renders', () => {
    const component = renderer.create(
      <Grommet>
        <CheckBoxGroup disabled options={['First', 'Second']} />
        <CheckBoxGroup options={[{ label: 'First', disabled: true }]} />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
