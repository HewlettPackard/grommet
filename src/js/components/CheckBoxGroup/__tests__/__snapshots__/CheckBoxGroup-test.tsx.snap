// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup custom theme 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 iUBVl StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 dwFxbl"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup defaultValue renders 1`] = `
<DocumentFragment>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jImhHp"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
      role="group"
    >
      <label
        class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
        label="First"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        >
          <input
            checked=""
            class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
            type="checkbox"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          >
            <svg
              class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
            >
              <path
                d="M6,11.3 L10.3,16 L18,6.2"
                fill="none"
              />
            </svg>
          </div>
        </div>
        <span>
          First
        </span>
      </label>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
      />
      <label
        class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
        label="Second"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        >
          <input
            class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
            type="checkbox"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          />
        </div>
        <span>
          Second
        </span>
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CheckBoxGroup disabled renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 kHUgtp"
      disabled=""
      label="First"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 iDmfOM"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 kHUgtp"
      disabled=""
      label="Second"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 iDmfOM"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          disabled=""
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 kHUgtp"
      disabled=""
      label="First"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 iDmfOM"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 kHUgtp"
      disabled=""
      label="First"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
        disabled=""
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 iDmfOM"
          disabled=""
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
          disabled=""
        />
      </div>
      <span>
        First
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup initial value renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="Maui"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        Maui
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="Jerusalem"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Jerusalem
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="Wuhan"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        Wuhan
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup labelKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onChange 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kgxPVP StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup onClick for check and uncheck a CheckBox 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kgxPVP StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
    tabindex="0"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup options renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="First"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="Second"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup value renders 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="First"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          checked=""
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jrmzgF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        >
          <svg
            class="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 caYxBM"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,11.3 L10.3,16 L18,6.2"
              fill="none"
            />
          </svg>
        </div>
      </div>
      <span>
        First
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="Second"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        Second
      </span>
    </label>
  </div>
</div>
`;

exports[`CheckBoxGroup valueKey 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 jImhHp"
>
  <div
    class="StyledBox-sc-13pk1d4-0 kAKeln StyledCheckBoxGroup-sc-2nhc5d-0"
    role="group"
  >
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="first-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        first-label
      </span>
    </label>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 fNTNlK"
    />
    <label
      class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 iVFNAQ"
      label="second-label"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 eUgRdO StyledCheckBox-sc-1dbk5ju-6 bWiGFS"
      >
        <input
          class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dxOFzI"
          type="checkbox"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 jMHXSN StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 gmfmIo"
        />
      </div>
      <span>
        second-label
      </span>
    </label>
  </div>
</div>
`;
